gobblin-data-management/src/main/java/gobblin/data/management/retention/dataset/MultiVersionCleanableDatasetBase.java

```diff
@@ -261,6 +283,14 @@ public void clean() throws IOException {
       Collection<T> deletableVersions = selectionPolicy.listSelectedVersions(versions);
 
       cleanImpl(deletableVersions);
+
+      List<DatasetVersion> allVersions = Lists.newArrayList();
+      for (T ver : versions) {
+        allVersions.add(ver);
+      }
+      for (RetentionAction retentionAction : versionFinderAndPolicy.getRetentionActions()) {
+        retentionAction.execute(allVersions);
```

Comment:
I think we should add some isolation (i.e. catching of exceptions) between the application of different actions.